core/solver/ir.cpp

```diff
@@ -160,21 +160,31 @@ std::unique_ptr<LinOp> Ir<ValueType>::conj_transpose() const
 
 template <typename ValueType>
 void Ir<ValueType>::apply_impl(const LinOp* b, LinOp* x) const
+{
+    this->apply_with_guess(b, x, this->get_apply_guess());
+}
+
+
+template <typename ValueType>
+void Ir<ValueType>::apply_with_guess(const LinOp* b, LinOp* x,
```

Comment:
I don't think we need to save on characters here: `guess` doesn't really have a consistent meaning for solvers, `initial_guess` does
```suggestion
void Ir<ValueType>::apply_with_initial_guess(const LinOp* b, LinOp* x,
```